tests/conftest.py

```diff
@@ -524,3 +525,37 @@ def _maker(
         return cast(Path, config_file.relative_to(tmpdir_cwd))
 
     return _maker
+
+
+@pytest.fixture(scope="session")
+def setuptools_wheel_path(tmp_path_factory):
+    """
+    Ensure setuptools is downloaded and return the path to the download directory.
+    """
+    tmp_path = tmp_path_factory.mktemp("tmp_wheels")
+    subprocess.check_call(
+        [
+            sys.executable,
+            "-m",
+            "pip",
+            "download",
+            "--exists-action=i",
+            f"--dest={tmp_path.as_posix()}",
+            "setuptools",
+        ],
+        stdout=subprocess.DEVNULL,
+    )
+    return tmp_path
+
+
+@pytest.fixture(scope="session")
+def minimal_wheels_path(setuptools_wheel_path):
+    """
+    Ensure minimal wheels and setuptools are downloaded and return the path.
+    """
+    for wheel_file in os.listdir(MINIMAL_WHEELS_PATH):
+        src = os.path.join(MINIMAL_WHEELS_PATH, wheel_file)
+        dst = os.path.join(setuptools_wheel_path, wheel_file)
+        if not os.path.exists(dst):
+            shutil.copy(src, dst)
```

Comment:
We should avoid using strings whenever possible, always preferring pathlib objects. So the low-level `os` module doesn't need to be used.
Also, the "if exists" check is unnecessary since it will never exist.